controllers/clusterresources/awssecuritygroupfirewallrule_controller.go

```diff
@@ -101,13 +102,13 @@ func (r *AWSSecurityGroupFirewallRuleReconciler) handleCreateFirewallRule(
 	if firewallRule.Status.ID == "" {
 		l.Info(
 			"Creating AWS security group firewall rule",
-			"cluster ID", firewallRule.Spec.ClusterID,
+			"cluster ID", firewallRule.Status.ClusterID,
 			"type", firewallRule.Spec.Type,
 		)
 
 		patch := firewallRule.NewPatch()
 
-		firewallRuleStatus, err := r.API.CreateFirewallRule(instaclustr.AWSSecurityGroupFirewallRuleEndpoint, &firewallRule.Spec)
+		firewallRuleStatus, err := r.API.CreateAWSSecurityGroupFirewallRule(instaclustr.AWSSecurityGroupFirewallRuleEndpoint, &firewallRule.Spec, firewallRule.Status.ClusterID)
```

Comment:
Form the endpoint inside API method please

Reply:
fixed